python/gigl/distributed/dataset_factory.py

```diff
@@ -86,6 +88,102 @@ def _get_labels_from_features(
     )
 
 
+# TODO (mkolodner-sc): Move this function to the partitioner once the partitioner accepts and returns
+# the node labels as separate fields from the node features.
+def _extract_node_labels_from_features(
+    partition_output: PartitionOutput,
+    feature_dim: Union[int, dict[NodeType, int]],
+) -> None:
+    """
+    Extract node labels from node features and set them in partition_output in-place, handling both heterogeneous and homogeneous cases.
+
+    Args:
+        partition_output (PartitionOutput): The partition output containing partitioned_node_features; will have partitioned_node_labels set
+        feature_dim (Union[int, dict[NodeType, int]]): Dimension of the node features.
+    """
+    node_labels: Optional[
+        Union[FeaturePartitionData, dict[NodeType, FeaturePartitionData]]
+    ] = None
+
+    label_dim: int
+
+    if isinstance(partition_output.partitioned_node_features, Mapping):
+        node_labels = {}
+        for (
+            node_type,
+            node_feature,
+        ) in partition_output.partitioned_node_features.items():
+            # feature_dim can be heterogeneous for a heterogeneous node features
+            if isinstance(feature_dim, Mapping):
+                feature_dim_for_current_node_type = feature_dim[node_type]
+            # feature_dim can be homogeneous for a heterogeneous node features, since we inject positive and negative label types as edges
+            else:
+                feature_dim_for_current_node_type = feature_dim
+            label_dim = node_feature.feats.shape[1] - feature_dim_for_current_node_type
+            if label_dim > 0:
+                (
+                    node_features_per_node_type,
+                    node_label_per_node_type,
```

Comment:
```suggestion
                    node_features_for_node_type,
                    node_label_for_node_type,
```

nit. "per" implies a dict to me, and these are just tensors right?

Reply:
For me this change is in and the code looks like: 
```
node_features_for_node_type,
                    node_label_for_node_type,
                ) = _get_labels_from_features(node_feature.feats, label_dim=label_dim)
```
Maybe refresh your tab?